### PR TITLE
Respect target CFLAGS. Added LDFLAGS.

### PR DESCRIPTION
**Pull request by xvitaly**
* Respect already defined by build environment CFLAGS.
* Added LDFLAGS.
* Fixed incorrect chmod on shared library and icons.

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -22,10 +23,10 @@ clean:
 
 install:
 	mkdir -p $(DESTDIR)$(PLUGIN_DIR_PURPLE)
-	install -m 664 $(TARGET) $(DESTDIR)$(PLUGIN_DIR_PURPLE)
+	install -m 0755 $(TARGET) $(DESTDIR)$(PLUGIN_DIR_PURPLE)
```
Comment (ignatenkobrain) [inline Makefile.common]: -p as well

Comment (richvdh) [inline Makefile.common]: why -p?

Comment (richvdh) [inline Makefile.common]: also: why 0755? it's a .so, not an executable.

File: Makefile.common [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # common parts of Makefile and Makefile.mingw
 
-COMPILE.c = $(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
+COMPILE.c = $(CC) $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
```
Comment (richvdh) [inline Makefile.common]: why is this useful? this is the rule for turning .c into .o, which is nothing to do with linking

File: Makefile.common (hunk 1 above)
Comment (ignatenkobrain) [inline Makefile.common]: because -p preserves timestamps

Comment (ignatenkobrain) [inline Makefile.common]: and 0755 because otherwise shared objects are not usable

File: Makefile.common (hunk 2 above)
Comment (xvitaly) [inline Makefile.common]: Yes, this line can be dropped.

**Comment by xvitaly** [comment]
I will close this pull request and open new #49 with fixes.

File: Makefile.common (hunk 1 above)
Comment (richvdh) [inline Makefile.common]: I know what -p does. Why do you want to do it?

shared libraries on Linux do not need +x.

**Pull request closed without merging**